pkg/yttlibrary/lib/api.go

```diff
@@ -0,0 +1,283 @@
+package lib
+
+import (
+	"fmt"
+
+	"github.com/k14s/ytt/pkg/eval"
+	"github.com/k14s/ytt/pkg/orderedmap"
+	"github.com/k14s/ytt/pkg/template/core"
+	"github.com/k14s/ytt/pkg/yamlmeta"
+	"github.com/k14s/ytt/pkg/yamltemplate"
+	"go.starlark.net/starlark"
+	"go.starlark.net/starlarkstruct"
+)
+
+func NewAPI(loader eval.Loader) starlark.StringDict {
+	return starlark.StringDict{
+		"lib": &starlarkstruct.Module{
+			Name: "lib",
+			Members: starlark.StringDict{
+				"get": starlark.NewBuiltin("lib.eval", core.ErrWrapper(libModule{loader}.Get)),
+			},
+		},
+	}
+}
+
+type libModule struct {
+	loader eval.Loader
+}
+
+func (m libModule) Get(
+	thread *starlark.Thread, f *starlark.Builtin,
+	args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+
+	if args.Len() != 1 {
+		return starlark.None, fmt.Errorf("expected one argument")
+	}
+
+	libPath, err := core.NewStarlarkValue(args.Index(0)).AsString()
+	if err != nil {
+		return starlark.None, err
+	}
+
+	l := &libraryToLoad{
+		loader:  m.loader,
+		libPath: libPath,
+		values:  []eval.ValuesAst{},
+	}
+
+	return l.starlark(), nil
+}
+
+type libraryToLoad struct {
+	loader  eval.Loader
+	libPath string
+	values  []eval.ValuesAst
+	result  *eval.Result
+	evalErr error
+}
+
+func (l *libraryToLoad) starlark() starlark.Value {
```

Comment:
starlark() -> AsStarlarkValue(); also should be public (even though object itself is private).